src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java

```diff
@@ -22,24 +38,42 @@
   private final Joystick m_driverJoystick;
   private final MotorController[] m_motorControllers;
   private final DifferentialDrive m_differentialDrive;
-  //public static ADIS16448_IMU m_gyro; Non-native gyro, might use later
-  public static ADXRS450_Gyro m_gyro;
+  private DifferentialDrivetrainSim m_DifferentialDrivetrainSim;
+  public final Field2d m_field = new Field2d();
+  private final double KvLinear = 1.98;
+  private final double KaLinear = 0.2;
+  private final double KvAngular = 1.5;
+  private final double KaAngular = 0.3;
+  private AnalogGyro m_gyro = new AnalogGyro(1);
+  private AnalogGyroSim m_gyroSim = new AnalogGyroSim(m_gyro);
+  //public static ADIS16448_IMU m_gyro; //Non-native gyro, might use later
+  //public static ADXRS450_Gyro m_gyro;
   private final DifferentialDriveOdometry m_odometry;
   public static Encoder m_leftEncoder;
   public static Encoder m_rightEncoder;
+  private EncoderSim m_leftEncoderSim;
+  private EncoderSim m_rightEncoderSim;
+  
 
   // Here are the encoders
   
   
 
-  public DriveBaseSubsystem(Joystick joystick) {  
+  public DriveBaseSubsystem(Joystick joystick) { 
     m_leftEncoder = new Encoder(Constants.leftEncoderDIOone, Constants.leftEncoderDIOtwo, 
-    false, Encoder.EncodingType.k2X);
+                                false, Encoder.EncodingType.k2X);
+
     m_rightEncoder = new Encoder(Constants.rightEncoderDIOone, Constants.rightEncoderDIOtwo, 
-    false, Encoder.EncodingType.k2X);
+                                false, Encoder.EncodingType.k2X);
+
+    m_leftEncoderSim = new EncoderSim(m_leftEncoder);
+    m_rightEncoderSim = new EncoderSim(m_rightEncoder);
+
     m_driverJoystick = joystick;
+    SmartDashboard.putData("Field", m_field);
```

Comment:
Move the m_field and simulated encoder lines to the Robot.isSimulation section

Reply:
done